Tasks/SonarQubePostTest/SonarQubeHelper.ps1

```diff
@@ -0,0 +1,230 @@
+###############################################################################################
+### Note: This file should be kept in sync with the SonarQubeHelper.ps1 from the "end task"####
```

Comment:
Hmm - I added comments about this in the first commit of this PR but I guess they were not persisted. All functions until Retry were copied from elsewhere in the tasks. Retry and the functions doing the REST calls are copied from the Appveyor tests. The rest are new.